client/Assets/Scripts/CharacterFeedbacks.cs

```diff
@@ -82,8 +79,25 @@ void Update()
             ChangeModelsOverlayColor(currentOverlayColor);
             currentOverlayColor = nextColor;
         }
+
+        PlayHapticDamageFeedback();
     }
 
+    private void PlayHapticDamageFeedback(){
+        Dictionary<ulong,ulong> damageDone = GameServerConnectionManager.Instance.damageDone;
+        foreach (var damage in damageDone){
+            if(damage.Key == GameServerConnectionManager.Instance.playerId){
+                HapticFeedbackType hapticFeedbackType = GetHapticTypeByDamage(damage.Value);
+                TriggerHapticFeedback(hapticFeedbackType);
+            }
+        }
```

Comment:
```suggestion
        Dictionary<ulong,ulong> damageDone = GameServerConnectionManager.Instance.damageDone;
        foreach (var damage in damageDone){
            if(damage.Key == GameServerConnectionManager.Instance.playerId){
                HapticFeedbackType hapticFeedbackType = GetHapticTypeByDamage(damage.Value);
                TriggerHapticFeedback(hapticFeedbackType);
            }
        }ulong damage;
Dictionary<ulong,ulong> damageDone = GameServerConnectionManager.Instance.damageDone;
ulong playerId = GameServerConnectionManager.Instance.playerId
if (!damageDone.TryGetValue(playerId, out damage)) {
                HapticFeedbackType hapticFeedbackType = 
                    GetHapticTypeByDamage(damage);
                TriggerHapticFeedback(hapticFeedbackType);
}
```

We could use dictionaries this way instead of iterating over every damage value

Reply:
I really like this, thank you!!